File: code/ch03/3.2.1.reverseList.java
public class Solution {
  public ListNode reverseList(ListNode head) {
    if (!head) return null;
    ListNode prev = null, cur = head;
    while (cur) {
      cur.next = prev;
      prev = cur;
      cur = cur.next;
    }
    return prev;
  }
}

File: code/ch03/3.2.1.reverseList.cpp
#include <cstdio>
#include <cstdlib>

class Solution {
public:
  ListNode *reverseList(ListNode *head) {
    if (!head) return NULL;
    ListNode *prev = NULL, *cur = head;
    while (cur) {
      cur->next = prev;
      prev = cur;
      cur = cur->next;
    }
    return prev;
  }
};
